docs/explanation/pebble.rst

```diff
@@ -0,0 +1,85 @@
+.. _pebble_explanation_page:
+
+Pebble
+======
+
+.. important::
+    **Pebble is the default entrypoint for all ROCKs!**
+
+Similar to other well-known process managers such as *supervisord*, *runit*, or
+*s6*, `Pebble`_ is a service manager that enables the seamless orchestration of
+a collection of local service processes as organised set. The main difference
+is that `Pebble`_ has been designed with custom-tailored features that
+significantly enhance the overall container experience, making it the ideal
+candidate for the container's init process (also known as the entrypoint,
+with PID=1).
+
+Multiple processes in a container?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Containers' best practices advocate the separation of concerns and the adoption
+of a single service per container. With the introduction of `Pebble`_ as the
+ROCKs' entrypoint, this principle is elevated to new heights:
+
+  *if multiple processes rely on shared dependencies and are tightly coupled
+  together (i.e. they serve a single purpose and cannot be executed
+  independently), then the best practice entails orchestrating them within the
+  same container, with `Pebble`_ as their manager.*
+
+This new notion addressed existing pain points arising from the excessive
+separation of concerns, which results in numerous container images whose
+entrypoints lack the ability to gracefully handle the underlying child
+processes. This is one of the main reasons behind the gradual shift in best
+practices, as there is an increasing emphasis on adopting init processes such
+as `tini`_, `s6-overlay`_, or `Pebble`_.
+
+What to expect?
+~~~~~~~~~~~~~~~
+
+* **client-server model behind a single binary**: Pebble is injected into
+  ROCKs as a single binary which acts both as a daemon and a client to itself;
+* **declarative service definition**: the Pebble service processes (or simply
```

Comment:
i see the point for the  ``cleaner`` but for the ``error-prone``can you detail more ?

Reply:
`cleaner` and `less error-prone` are both advantages of a declarative format. with an imperative format, it is much easier to miss a symbol or a character in your script, causing the entire entrypoint to misbehave.